### DISCUSSION TITLE
How to build @excalidraw/excalidraw package

### DISCUSSION THREAD

**Discussion by kristiankostecky** [excalidraw/excalidraw#8207] [Q&A]
Hello everyone : ) 

I created a fork of excalidraw/excalidraw repository. I would like to modify @excalidraw/excalidraw react package and build it. I have a problem with building it. When I check the files from official package [in npm](https://www.npmjs.com/package/@excalidraw/excalidraw?activeTab=code) or check the node_modules there are those files/folder
```
dist/
types/ 
README.md
main.js
package.json
```

This differ from what I got when i run `yarn workspace @excalidraw/excalidraw build:esm`. I don't have `/types` folder only `/dist` and the content does not match. Am I missing some steps?

**Comment by dwelle** [comment]
Hi. The latest package is an ESM build, and corresponds to `0.17.1-xxxxxxx` versions: https://www.npmjs.com/package/@excalidraw/excalidraw/v/0.17.1-744b3e5?activeTab=code

**Reply by jiweiyuan** [reply]
Hi, why 0.17.1 is newer than 0.17.6?

**Reply by dwelle** [reply]
We're publishing the `@next` version right from the `release` branch and using `package.json` version as base, suffixing it with the commit hash (the 0.17.6 is a patch that's not merged in `master`).

I've created an issue to use the upcoming version instead so it's less confusing. https://github.com/excalidraw/excalidraw/issues/8215
